plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsToolkitExplorerFactory.kt

```diff
@@ -116,18 +137,31 @@ class AwsToolkitExplorerFactory : ToolWindowFactory, DumbAware {
                     if (ToolkitConnectionManager.getInstance(project)
                             .connectionStateForFeature(CodeCatalystConnection.getInstance()) == BearerTokenAuthState.AUTHORIZED
                     ) {
-                        showExplorerTree(project)
+                        loadContent(AwsToolkitExplorerToolWindow.getInstance(project), toolkitPanel)
                     }
                 }
             }
         )
+
+        project.messageBus.connect().subscribe(
+            ShowToolkitListener.TOPIC,
+            object : ShowToolkitListener {
+                override fun showWebview(project: Project) {
+                    loadContent(ToolkitWebviewPanel.getInstance(project).component, toolkitPanel)
+                }
+
+                override fun showExplorerTree(project: Project) {
+                    loadContent(AwsToolkitExplorerToolWindow.getInstance(project), toolkitPanel)
+                }
+            }
+        )
```

Comment:
same comment about not needing the messagebus here

Reply:
Will address it in a follow-up PR